src/Projection.cpp

```diff
@@ -0,0 +1,73 @@
+
+#include <ZividPython/Projection.h>
+
+#include <Zivid/Experimental/Projection.h>
+#include <Zivid/Resolution.h>
+#include <ZividPython/ReleasableCamera.h>
+#include <ZividPython/ReleasableProjectedImage.h>
+
+#include <array>
+#include <vector>
+
+namespace ZividPython::Projection
+{
+    void wrapAsSubmodule(pybind11::module &dest)
+    {
+        dest.def("projector_resolution", [](const ReleasableCamera &camera) {
+            const auto resolution = Zivid::Experimental::Projection::projectorResolution(camera.impl());
+            return std::make_pair(resolution.height(), resolution.width());
+        });
+
+        dest.def(
+            "show_image",
+            [](ReleasableCamera &camera,
+               const pybind11::array_t<uint8_t, pybind11::array::c_style | pybind11::array::forcecast> imageBGRA) {
+                const auto info = imageBGRA.request();
+
+                if(info.ndim != 3)
+                {
+                    throw std::runtime_error("Input image array must be three dimensional.");
+                }
+
+                const auto height = info.shape[0];
+                const auto width = info.shape[1];
+                const auto channels = info.shape[2];
+
+                if(channels != 4)
+                {
+                    throw std::runtime_error("Input image array must have four color channels (BGRA).");
+                }
+
+                const auto resolution = Zivid::Resolution{ static_cast<size_t>(width), static_cast<size_t>(height) };
+                const auto *dataPtr = reinterpret_cast<const Zivid::ColorBGRA *>(imageBGRA.data());
+                const Zivid::Image<Zivid::ColorBGRA> zividImage{ resolution, dataPtr, dataPtr + resolution.size() };
```

Comment:
It would be better to use the Image constructor that takes unsigned char begin/end ptrs, to avoid strict aliasing rule problems (this constructor was added quite late in 2.10 development so it likely didnt exist when you wrote this).

Reply:
Ah yes, I'll give it a shot.

Reply:
Like this? Seems to work fine.
https://github.com/zivid/zivid-python/pull/234/commits/1a1eb23f0e102259c0a94f28eafb4f5c021028db